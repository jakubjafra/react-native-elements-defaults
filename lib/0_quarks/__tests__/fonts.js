describe('with no theme applied', () => {
    let OldFonts, Fonts;

    beforeAll(() => {
        let defaultStyles = {};

        OldFonts = require('react-native-elements/src/config/fonts').default;
        Fonts = require('../fonts').compose(defaultStyles)(OldFonts);

        console.log(OldFonts, Fonts);
    });

    test('should define some default colors', () => {
        expect(OldFonts).toMatchSnapshot();
    });

    test('there should be default colors assigned to new Fonts', () => {
        expect(OldFonts).toEqual(Fonts);
    });
});

describe('with basic theme applied', () => {
    let OldFonts, Fonts;

    beforeAll(() => {
        let defaultStyles = {
            android: {
                regular: {
                    fontFamily: 'serif'
                }
            }
        };

        OldFonts = require('react-native-elements/src/config/fonts').default;
        Fonts = require('../colors').compose(defaultStyles)(OldFonts);
    });

    test('it should override the default color', () => {
        expect(Fonts.android.regular.fontFamily).toEqual('serif');
        expect(Fonts).toMatchSnapshot();
    });

    test('there should be default colors assigned to new Fonts', () => {
        expect(OldFonts).not.toEqual(Fonts);
    });
});
