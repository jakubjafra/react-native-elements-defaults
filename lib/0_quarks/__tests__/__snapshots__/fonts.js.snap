// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with basic theme applied it should override the default color 1`] = `
Object {
  "android": Object {
    "regular": Object {
      "fontFamily": "serif",
    },
  },
  "ios": Object {},
}
`;

exports[`with no theme applied should define some default colors 1`] = `
Object {
  "android": Object {
    "black": Object {
      "fontFamily": "sans-serif",
      "fontWeight": "bold",
    },
    "bold": Object {
      "fontFamily": "sans-serif",
      "fontWeight": "bold",
    },
    "condensed": Object {
      "fontFamily": "sans-serif-condensed",
    },
    "condensed_light": Object {
      "fontFamily": "sans-serif-condensed",
      "fontWeight": "light",
    },
    "light": Object {
      "fontFamily": "sans-serif-light",
    },
    "medium": Object {
      "fontFamily": "sans-serif-medium",
    },
    "regular": Object {
      "fontFamily": "sans-serif",
    },
    "thin": Object {
      "fontFamily": "sans-serif-thin",
    },
  },
  "ios": Object {},
}
`;
